command/genesis/polybft_params.go

```diff
@@ -98,7 +98,7 @@ func (p *genesisParams) generatePolyBftChainConfig(o command.OutputFormatter) er
 		rewardTokenAddr     = contracts.NativeERC20TokenContract
 	)
 
-	if p.rewardTokenCode == "" {
+	if p.rewardTokenCode == "" && p.epochReward > 0 {
```

Comment:
If epoch reward is 0, this will go to `else` condition, which will fail.

I suggest this:

```
if p.rewardTokenCode == "" {
      if p.epochReward > 0 {
            // native token is used as a reward token, and reward wallet is not a zero address
	    // so we need to add that address to premine map
	    premineBalances[walletPremineInfo.address] = walletPremineInfo
      }
} else {
      bytes, err := hex.DecodeString(p.rewardTokenCode)
      if err != nil {
	   return fmt.Errorf("could not decode reward token byte code '%s' : %w", p.rewardTokenCode, err)
      }

      rewardTokenByteCode = bytes
      rewardTokenAddr = contracts.RewardTokenContract
}
```